tests/mocha/contextmenu_items_test.js

```diff
@@ -0,0 +1,430 @@
+        // Mocks the confirmation dialog and calls the callback with 'true' simulating ok.
+        sinon.stub(Blockly, 'confirm').callsArgWith(1, false);
+
+        workspace.newBlock('text');
+        workspace.newBlock('text');
+        deleteOption.callback(scope);
+        this.clock.runAll();
+        chai.assert.equal(workspace.getTopBlocks(false).length, 2);
+      });
+
+      test('No dialog for single block', function() {
+        var confirmStub = sinon.stub(Blockly, 'confirm');
+        workspace.newBlock('text');
+        deleteOption.callback(scope);
+        this.clock.runAll();
+
+        sinon.assert.notCalled(confirmStub);
+        chai.assert.equal(workspace.getTopBlocks(false).length, 0);
+      });
+
+      test('Has correct label for multiple blocks', function() {
+        workspace.newBlock('text');
+        workspace.newBlock('text');
+
+        chai.assert.equal(deleteOption.displayText(scope), 'Delete 2 Blocks');
+      });
+
+      test('Has correct label for single block', function() {
+        workspace.newBlock('text');
+        chai.assert.equal(deleteOption.displayText(scope), 'Delete Block');
+      });
+    });
+  });
+
+  suite('Block Items', function() {
+    var scope;
+    var block;
+
+    setup(function() {
+      block = workspace.newBlock('text');
+      scope = {block: block};
+    });
+
+    teardown(function() {
+      block.dispose();
+    });
+
+    suite('Duplicate', function() {
+      var duplicateOption;
+      
+      setup(function() {
+        duplicateOption = registry.getItem('blockDuplicate');
+      });
+
+      test('Enabled when block is duplicatable', function() {
+        // Block is duplicatable by default
+        chai.assert.equal(duplicateOption.preconditionFn(scope), 'enabled');
+      });
+
+      test('Disabled when block is not dupicatable', function() {
+        sinon.stub(block, 'isDuplicatable').returns(false);
+        chai.assert.equal(duplicateOption.preconditionFn(scope), 'disabled');
+      });
+
+      test('Hidden when in flyout', function() {
+        block.isInFlyout = true;
+        chai.assert.equal(duplicateOption.preconditionFn(scope), 'hidden');
+      });
+
+      test('Calls duplicate', function() {
+        var stub = sinon.stub(Blockly, 'duplicate');
+
+        duplicateOption.callback(scope);
+
+        sinon.assert.calledOnce(stub);
+        sinon.assert.calledWith(stub, block);
+      });
+
+      test('Has correct label', function() {
+        chai.assert.equal(duplicateOption.displayText(), 'Duplicate');
+      });
+    });
+
+    suite('Comment', function() {
+      var commentOption;
+
+      setup(function() {
+        commentOption = registry.getItem('blockComment');
+      });
+
+      test('Enabled for normal block', function() {
+        chai.assert.equal(commentOption.preconditionFn(scope), 'enabled');
+      });
+
+      test('Hidden for IE', function() {
+        Blockly.utils.userAgent.IE = true;
```

Comment:
This is changing a global variable and since we don't want this to affect any other tests, I think maybe add something like:
```
var oldState = Blockly.utils.userAgent.IE;
try {
Blockly.utils.userAgent.IE = true;
chai.assert.equal(commentOption.preconditionFn(scope), 'hidden');
} finally {
Blockly.utils.userAgent.IE = oldState;
}
```

Reply:
Thanks, done